### ISSUE TITLE
Question: Why we are using gRPC?

### ISSUE THREAD

**Issue by alexrios** [stone-co/the-amazing-ledger#26]
I don't know the motivation 😞 .

Since gRPC will definitely be part of the project I strongly suggest to stop maintaining a tailor-made HTTP API and to adopt [grpc gateway](https://github.com/grpc-ecosystem/grpc-gateway).

The upside is to reduce boilerplate and avoid discussions like: _What is the best status code for this situation_?

What you guys think?

**Comment by alexgarzao** [comment]
I never used the grpc gateway. I took a look now, and it seems to be a good option.

What do you think @juceliofloresta ?

**Comment by juceliofloresta** [comment]
I never used grpc too.

I think that uses gRPC and HTTP can be increase number of partners that work with Ledger. 

Maybe the technology choose don't have integration with RPC, they can use HTTP to integrate service with Ledger.

**Comment by alexgarzao** [comment]
Take a look at https://github.com/grpc-ecosystem/grpc-gateway @juceliofloresta 

The proposal was not to remove the HTTP API, but to use grpc gateway that can provide both interfaces (RPC and HTTP).

**Comment by alexrios** [comment]
I see. 

So if I got the idea, we are using gRPC as default communication mechanism and keep the HTTP API (_REST-ish_) as a fallback option.

The gRPC-gateway will provide one single source of truth as something like this:
```go
rpc GetTransaction(GetTransactionsRequest) returns (GetTransactionsResponse) {
    option (google.api.http) = {
      get: "/api/v1/transactions"
    };
  }
```

This way we have HTTP and gRPC (plus swagger) implementation autogenerated and bring more focus on use cases.

**Comment by juceliofloresta** [comment]
LGTM

Using google.api.http annotation in proto file, the generated [pb](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/echo_service.pb.go) file have json tags in structs.

> The gRPC-gateway will provide one single source of truth as something like this

 I like that, so the proto file is source of truth and we don't mistakes with isolated sources.

I think that we need [start grpc server to register grpc endpoint](https://github.com/grpc-ecosystem/grpc-gateway#usage) in http server.

**Comment by alexrios** [comment]
Just for make it clear, are you guys planning to do it internally or we should open an issue?

**Comment by alexgarzao** [comment]
At this time, we are developing some issues that are not at Github. 

We need to put all issues at GitHub to be more visible for everyone. Sorry, it was my fault.

Please, open an issue. If someone can address this, nice. If not, we will address ASAP.

I will try to put all issues on GitHub ASAP.

**Comment by alexrios** [comment]
it will be resolved on #27.
